api.go

```diff
@@ -696,6 +700,16 @@ func (r *Raft) ReloadConfig(rc ReloadableConfig) error {
 		return err
 	}
 	r.conf.Store(newCfg)
+
+	if rc.HeartbeatTimeout < oldCfg.HeartbeatTimeout {
+		// On leader, ensure replication loops running with a longer
+		// timeout than what we want now discover the change.
+		for _, repl := range r.leaderState.replState {
+			asyncNotifyCh(repl.notifyCh)
+		}
```

Comment:
Is it safe to access `leaderState` here outside the leader loop? I assume this read could potentially race with an update to that state from the leader loop couldn't it?

Reply:
Ah, good catch, probably not.  Let me see if I can re-work.

Reply:
Pushed a fix.